pkg/controller/handler.go

```diff
@@ -230,8 +230,9 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 
 		// ExecutionNotFound error is returned when flyteadmin is missing the workflow. This is not
 		// a valid state unless we are experiencing a race condition where the workflow has not yet
-		// been inserted into the db (ie. workflow phase is WorkflowPhaseReady).
-		if err != nil && eventsErr.IsNotFound(err) && w.GetExecutionStatus().GetPhase() != v1alpha1.WorkflowPhaseReady {
+		// been inserted into the db. Therefore, we allow retries until the maximum number of
+		// failed attempts is reached.
+		if err != nil && eventsErr.IsNotFound(err) && w.Status.FailedAttempts > uint32(p.cfg.MaxWorkflowRetries) {
```

Comment:
Won't [this](https://github.com/flyteorg/flytepropeller/blob/c22158126a1785165ab97d1ab83eaf063e244b66/pkg/controller/handler.go#L95) eventually break out anyway?

Reply:
Yes, you are correct. This is not the right fix for this, I have done some extensive testing now and found the issue was a cyclic ExecutionNotFound error when attempted to abort the workflow. I will leave this as is.